MvvmCross/Windows/Wpf/Views/Presenters/MvxWpfViewPresenter.cs

```diff
@@ -0,0 +1,202 @@
+﻿// MvxWpfViewPresenter.cs
+
+// MvvmCross is licensed using Microsoft Public License (Ms-PL)
+// Contributions and inspirations noted in readme.md and license.txt
+//
+// Project Lead - Stuart Lodge, @slodge, me@slodge.com
+
+using MvvmCross.Core.ViewModels;
+using MvvmCross.Core.Views;
+using MvvmCross.Platform;
+using MvvmCross.Platform.Platform;
+using MvvmCross.Wpf.Views.Presenters.Attributes;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Windows;
+
+namespace MvvmCross.Wpf.Views.Presenters
+{
+    public class MvxWpfViewPresenter
+        : MvxBaseWpfViewPresenter
+    {
+        private Dictionary<Type, Action<FrameworkElement, MvxBasePresentationAttribute, MvxViewModelRequest>> _attributeTypesToShowMethodDictionary;
+        protected Dictionary<Type, Action<FrameworkElement, MvxBasePresentationAttribute, MvxViewModelRequest>> AttributeTypesToShowMethodDictionary
+        {
+            get
+            {
+                if (_attributeTypesToShowMethodDictionary == null)
+                {
+                    _attributeTypesToShowMethodDictionary = new Dictionary<Type, Action<FrameworkElement, MvxBasePresentationAttribute, MvxViewModelRequest>>();
+                    RegisterAttributeTypes();
+                }
+                return _attributeTypesToShowMethodDictionary;
+            }
+        }
+
+        private readonly Dictionary<Window, Stack<FrameworkElement>> _frameworkElementsDictionary = new Dictionary<Window, Stack<FrameworkElement>>();
+
+        public MvxWpfViewPresenter(Window window)
+        {
+            _frameworkElementsDictionary.Add(window, new Stack<FrameworkElement>());
+        }
+
+        protected virtual void RegisterAttributeTypes()
+        {
+            _attributeTypesToShowMethodDictionary.Add(
+                typeof(MvxWindowPresentationAttribute),
+                (element, attribute, request) => ShowWindow(element, (MvxWindowPresentationAttribute)attribute, request));
+
+            _attributeTypesToShowMethodDictionary.Add(
+                typeof(MvxContentPresentationAttribute),
+                (element, attribute, request) => ShowContentView(element, (MvxContentPresentationAttribute)attribute, request));
+        }
+
+        public override void ChangePresentation(MvxPresentationHint hint)
+        {
+            base.ChangePresentation(hint);
+
+            if (hint is MvxClosePresentationHint)
+            {
+                Close((hint as MvxClosePresentationHint).ViewModelToClose);
+            }
+        }
+
+        public override void Show(MvxViewModelRequest request)
+        {
+            var loader = Mvx.Resolve<IMvxWpfViewLoader>();
```

Comment:
Maybe it's better to add a backing field for the `IMvxWpfViewLoader` instead of resolving every time.